### ISSUE TITLE
Bad interaction between fixed_params sampler and num_chains argument

### ISSUE THREAD

**Issue by WardBrian** [stan-dev/cmdstan#1057]
#### Summary:

When `algorithm=fixed_param` (either implicitly as is being removed in #1054  or explicitly) and num_chains is set to a value other than `1`, multiple files are created but only the first contains any draws.

#### Description:
Describe the issue as clearly as possible.

#### Reproducible Steps:
Model:
```stan
data { 
  int<lower=0> N;
  real<lower=0,upper=1> theta;
} 
generated quantities {
  int theta_rep = 0;
  for (n in 1:N)
    theta_rep += bernoulli_rng(theta);
}
```

data:
```
{
"N":300,
"theta": 0.3
}
```

command:
`./bernoulli_datagen sample num_chains=7 algorithm=fixed_param data file=data.json`

#### Current Output:

Files `output_#.csv` for 1 to 7. File 1 contains draws, remainder contain only the CmdStan header comment with no csv data.

#### Expected Output:
Either this should produce an error or each file should contain draws

#### Additional Information:
Related:
https://github.com/stan-dev/cmdstanpy/issues/495

#### Current Version:
v2.28.1

**Comment by rok-cesnovar** [comment]
num_chains is currently only supported with NUTS + dense_e or diag_e. Anything else is not supported. A helpful error message would be nice though.

**Comment by WardBrian** [comment]
Silently ignoring that it was requested is pretty bad, yeah. The fact that it outputs 7 files still is even worse, especially when combined with the current behavior that fixed_params can be turned on without the user asking for it

**Comment by nsiccha** [comment]
Same problem for nuts without adaptation.

**Comment by rok-cesnovar** [comment]
@nsiccha so your model has parameters and you set `adapt engaged=0`? I tried that with the bernoulli example and it seems to work fine. Can you provide a bit more detail?

**Comment by nsiccha** [comment]
@rok-cesnovar Yes, `num_chains > 1` and `sample adapt engaged=0` with the bernoulli example behaves exactly the same way as Brian describes above: 

Multiple files are created, only the first contains draws and no error is reported.

I doubt that it can work for you. 

First, there is a hidden error message describing this lacking feature but which doesn't get printed here: https://github.com/stan-dev/cmdstan/blob/34145a3f9f632d6d09fd229e4cb4768869ae7f3e/src/cmdstan/command.hpp#L359-L361
Second, the calls to non-adapting nuts extract only the first "context", see e.g. here:
https://github.com/stan-dev/cmdstan/blob/34145a3f9f632d6d09fd229e4cb4768869ae7f3e/src/cmdstan/command.hpp#L815-L819
Finally, there's simply no mention at all of num_chains at all outside of the adapting nuts calls:
https://github.com/stan-dev/stan/search?q=num_chains+path%3Asrc%2Fstan%2Fservices

**Comment by rok-cesnovar** [comment]
Ah yes, this is simply because `num_chains > 1` is not supported when using NUTS without adaptation. Not sure why the throw is not registered though. This is annoying for sure.

**Comment by nsiccha** [comment]
Well I'll simply not use it till everything is ready, so no big deal for me. :shrug:

**Comment by rok-cesnovar** [comment]
That sounds reasonable. The error messaging will definitely be improved for the next release, not sure on providing multiple chains within a single executable for all services. That will probably be extended more slowly.

**Comment by WardBrian** [comment]
Updated the title to reflect that this isn't just a fixed_param issue but any unsupported sampler

**Comment by WardBrian** [comment]
@rok-cesnovar will this be improved before 2.29?

**Comment by rok-cesnovar** [comment]
Yeah, should have time to expand error messaging this week.
